// chai/lib/chai/utils/index.js
/*!
 * chai
 * Copyright(c) 2011 Jake Luer <jake@alogicalparadox.com>
 * MIT Licensed
 */

/*!
 * Dependencies that are used for multiple exports are required here only once
 */

var pathval = require('pathval');

/*!
 * test utility
 */

exports.test = require('chai/lib/chai/utils/test');

/*!
 * type utility
 */

exports.type = require('type-detect');

/*!
 * expectTypes utility
 */
exports.expectTypes = require('chai/lib/chai/utils/expectTypes');

/*!
 * message utility
 */

exports.getMessage = require('chai/lib/chai/utils/getMessage');

/*!
 * actual utility
 */

exports.getActual = require('chai/lib/chai/utils/getActual');

/*!
 * Inspect util
 */

exports.inspect = require('chai/lib/chai/utils/inspect');

/*!
 * Object Display util
 */

exports.objDisplay = require('chai/lib/chai/utils/objDisplay');

/*!
 * Flag utility
 */

exports.flag = require('chai/lib/chai/utils/flag');

/*!
 * Flag transferring utility
 */

exports.transferFlags = require('chai/lib/chai/utils/transferFlags');

/*!
 * Deep equal utility
 */

exports.eql = require('deep-eql');

/*!
 * Deep path info
 */

exports.getPathInfo = pathval.getPathInfo;

/*!
 * Check if a property exists
 */

exports.hasProperty = pathval.hasProperty;

/*!
 * Function name
 */

exports.getName = require('get-func-name');

/*!
 * add Property
 */

exports.addProperty = require('chai/lib/chai/utils/addProperty');

/*!
 * add Method
 */

exports.addMethod = require('chai/lib/chai/utils/addMethod');

/*!
 * overwrite Property
 */

exports.overwriteProperty = require('chai/lib/chai/utils/overwriteProperty');

/*!
 * overwrite Method
 */

exports.overwriteMethod = require('chai/lib/chai/utils/overwriteMethod');

/*!
 * Add a chainable method
 */

exports.addChainableMethod = require('chai/lib/chai/utils/addChainableMethod');

/*!
 * Overwrite chainable method
 */

exports.overwriteChainableMethod = require('chai/lib/chai/utils/overwriteChainableMethod');

/*!
 * Compare by inspect method
 */

exports.compareByInspect = require('chai/lib/chai/utils/compareByInspect');

/*!
 * Get own enumerable property symbols method
 */

exports.getOwnEnumerablePropertySymbols = require('chai/lib/chai/utils/getOwnEnumerablePropertySymbols');

/*!
 * Get own enumerable properties method
 */

exports.getOwnEnumerableProperties = require('chai/lib/chai/utils/getOwnEnumerableProperties');

/*!
 * Checks error against a given set of criteria
 */

exports.checkError = require('check-error');

/*!
 * Proxify util
 */

exports.proxify = require('chai/lib/chai/utils/proxify');

/*!
 * addLengthGuard util
 */

exports.addLengthGuard = require('chai/lib/chai/utils/addLengthGuard');

/*!
 * isProxyEnabled helper
 */

exports.isProxyEnabled = require('chai/lib/chai/utils/isProxyEnabled');

/*!
 * isNaN method
 */

exports.isNaN = require('chai/lib/chai/utils/isNaN');

/*!
 * getOperator method
 */

exports.getOperator = require('chai/lib/chai/utils/getOperator');
